rfcs/2021-01-05-typography-scaling.md

```diff
@@ -0,0 +1,221 @@
+
+A `height-factor` is the unit which tells you that how many times the font-size should be multiplied by it to get actual `line-height`.
+
+For example,
+```
+.headingXXL-Desktop {
+  line-height: 60px; /* (1.5) looks good on large screens */
+  font-size: 40px;
+}
+
+.headingXXL-Mobile {
+  line-height: 38px; /* (1.3) looks good on small screens */
+  font-size: 28px;
+}
+```
+
+General Formula,
+
+>**`line-height = font-size * height-factor`**
+
+And will round-off the resulting value.
+
+### **Storage of Units**
+**Global Level Tokens:** All the font attributes such as `font-family`, `line-height`, `font-size` & `font-weight` will be stored as global-level tokens.
+
+**Alias-Level Tokens:** Text colors will be stored as a alias/theme-level tokens.
+
+**Component-Level Tokens:** The main typography-scale such as `headings`, `body`, etc. will be stored as component-level tokens.
+
+### **Current Limitations**
+There are two major limitations while scaling typography to multiple breakpoints,
+1. **Need of a different typography scale.** If any product wants to bring in their own typography scale then they need to override the current typography scale (without addition/subtraction of any levels) provided by blade. This also means that you cannot add/remove levels from the base scale. This is currently being looked at by the comm. design team.
+2. **Same type scaling across all devices/breakpoints.** After the type scale is defined, we also need to define the scale across multiple device sizes. There are few approaches that we've came up with and after research from other DS/products like *GitHub*, *Eightshapes*, *Paystack*, *Stripe*, and a few more we've observed that for dashboards generally tweaking the **heading** sizes works best and the base font-size will remain same for all other devices/breakpoints.
+
+We'll look into the responsive type scale and see what approach fits best for our use case.
+
+>📝 Note: **Responsive Typography** will scale in segments. Which means if `h1` at `1440px` is `40px`, then at `991px` it will become `32px` directly rather than going from `40px -> 38px -> 36px -> 34px -> 32px...`.
+
+### **Our Approach**
+>**Multiple type-scaling across breakpoints**
+
+Our product is majorly used on mobiles/tablets/desktops. Therefore, We'll  consider only two breakpoints for the typography scale, one for mobiles & one for both desktops & tablets. Which is for devices having width `>=991px` (approx.)
+
+![Scaling: Multiple type scales ratio](./images/typography/typography-multi-type-scaling.png)
+
+In this type of scaling the modular scale changes on every breakpoint. For example: **mobiles** will use `Minor Third (1.2)` and **desktops/tablets** will use `Custom Scale (1.1)`.
```

Comment:
```suggestion
In this type of scaling the modular scale changes on every breakpoint. For example: **desktops/tablets** will use `Minor Third (1.2)` and **mobiles** will use `Custom Scale (1.1)`.
```
typo 😄

Reply:
And instead of `Minor Second (1.2)` it will be `Major Second (1.125)` 😅

Reply:
How is this determined? Major / minor, didn't find any context in the doc, or maybe I missed it.

Reply:
This has been picked up visually, there is a little context we've mentioned. As for the tools like the dashboard, it is better to use `Major Second` as we get **more control and flexibility of choosing font sizes** with this approach.